cfpb/consumerfinance.gov
# Improve filterable list controls a11y

Add a `group` role to the filterable list controls checkboxes parents so that screenreaders announce the group name when navigating to the checkbox children.

Normally our use of `<legend>` and `<label>` would do this automatically but we nest the checkboxes inside an unordered list so we need to manually tell the browser that the items are grouped.

See https://github.local/CFGOV/platform/issues/3356 for background.

## Additions

- `role` attribute to the category and status unordered lists.

## How to test this PR

1. Start voiceover.
1. Go to http://0.0.0.0:8000/about-us/blog/
1. Use the tab key to navigate to the `Category` checkboxes in the filter controls.
1. It should announce "Category group" after announcing the first checkbox item.
1. Go to https://www.consumerfinance.gov/about-us/blog/ and notice that it doesn't announce the group name.

## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
